sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1920,7 +1927,34 @@ class Analyzer(
 
       case p: LogicalPlan if !p.childrenResolved => p
 
-      // Aggregate without Having clause.
+      // Extract window expressions from aggregate functions. There might be an aggregate whose
+      // aggregate function contains a window expression as a child, which we need to extract.
+      // e.g., df.groupBy().agg(max(rank().over(window))
+      case a @ Aggregate(groupingExprs, aggregateExprs, child)
```

Comment:
To be sure: What happens if there is a window function on-top of the aggregate function? This gets resolved in two passes right?

Reply:
Can you add a test case for this scenario?